### PR TITLE
2PC-MPC code cleanup

### PR DESCRIPTION
**Pull request by zeevmoney**

### CODE REVIEW

File: crates/pera-core/src/authority.rs [hunk 1]
```diff
@@ -1548,47 +1546,51 @@ impl AuthorityState {
...
-                dwallet_mpc_outputs_manager.network_key_version(),
-            ) else {
-                continue;
+                dwallet_mpc_outputs_verifier.network_key_version(),
+            ) {
+                Ok(Some(info)) => info,
+                _ => continue,
             };
+
+            // todo(zeev): call batches manager only if it's a batch in the first place.
```
Comment (zeevmoney) [inline crates/pera-core/src/authority.rs:1578]:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

File: crates/pera-core/src/authority/authority_per_epoch_store.rs [hunk 2]
```diff
@@ -1037,43 +1035,37 @@ impl AuthorityPerEpochStore {
...
-                    ))
-                })
-                .collect::<Result<HashMap<_, _>, _>>()?,
-        })
+                .map(|(k, v)| Ok((*k, bcs::from_bytes::<ClassGroupsPublicKeyAndProof>(v)?)))
+                .collect::<PeraResult<HashMap<AuthorityName, ClassGroupsPublicKeyAndProof>>>(),
+        }
     }
 
+    // todo(zeev): change to network decryption key share (also in the error).
```
Comment (zeevmoney) [inline crates/pera-core/src/authority/authority_per_epoch_store.rs:1047]:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

File: crates/pera-core/src/authority/authority_per_epoch_store.rs [hunk 3]
```diff
@@ -1037,43 +1035,37 @@ impl AuthorityPerEpochStore {
...
-        Ok(encryption_of_decryption_key_shares)
+        })
+        .ok_or(DwalletMPCError::MissingEncryptionOfDecryptionKeyShares)?
+        .contents
+        .into_iter()
+        .map(|entry| (entry.key, entry.value))
+        .collect())
     }
 
+    // todo(zeev): docs.
```
Comment (zeevmoney) [inline crates/pera-core/src/authority/authority_per_epoch_store.rs:1068]:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

File: crates/pera-core/src/dwallet_mpc/batches_manager.rs [hunk 4]
```diff
@@ -1,3 +1,6 @@
+// todo(zeev): module level doc.
```
Comment (zeevmoney) [inline crates/pera-core/src/dwallet_mpc/batches_manager.rs:1]:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

File: crates/pera-core/src/dwallet_mpc/mpc_instance.rs [hunk 5]
```diff
@@ -16,28 +16,38 @@ use crate::authority::authority_per_epoch_store::AuthorityPerEpochStore;
...
     pub(super) session_info: SessionInfo,
-    /// The MPC party that being used to run the MPC cryptographic steps. An option because it can be None before the instance has started.
-    pub party: MPCParty,
-    pub(super) public_input: Vec<u8>,
+    /// The MPC party being used to run the MPC cryptographic steps.
+    /// Party in here is not a Validator, but a cryptographic party.
+    party: MPCParty,
+    pub(super) public_input: MPCPublicInput,
     /// The decryption share of the party for mpc sign sessions
+    /// todo(zeev): why is it here? it's never used.
```
Comment (zeevmoney) [inline crates/pera-core/src/dwallet_mpc/mpc_instance.rs:43]:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

File: crates/pera-core/src/dwallet_mpc/mpc_instance.rs [hunk 6]
```diff
@@ -141,23 +151,27 @@ impl DWalletMPCInstance {
         }
     }
 
+    // todo(zeev): WHAT? WHY? DOCS.
```
Comment (zeevmoney) [inline crates/pera-core/src/dwallet_mpc/mpc_instance.rs:154]:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

File: crates/pera-core/src/dwallet_mpc/mpc_manager.rs [hunk 7]
```diff
@@ -55,29 +51,30 @@ pub struct DWalletMPCManager {
...
+    Message(MPCMessage, AuthorityName, ObjectID),
+    /// An output for a completed MPC message.
+    Output(MPCOutput, AuthorityName, SessionInfo),
+    /// A new session event.
     Event(Event, SessionInfo),
-    /// A signal that the delivery of messages has ended, now the instances that received a quorum of messages can advance
+    /// Signal delivery of messages has ended,
+    /// now the instances that received a quorum of messages can advance.
     EndOfDelivery,
+    // todo(zeev): doc this.
```
Comment (zeevmoney) [inline crates/pera-core/src/dwallet_mpc/mpc_manager.rs:77]:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

File: crates/pera-core/src/dwallet_mpc/mpc_manager.rs [hunk 8]
```diff
@@ -321,13 +319,14 @@ impl DWalletMPCManager {
...
 
-        // Need to send the messages one by one, so the consensus adapter won't think they
+        // Need to send the messages' one by one, so the consensus adapter won't think they
         // are a [soft bundle](https://github.com/sui-foundation/sips/pull/19).
         for message in messages {
             self.consensus_adapter
                 .submit_to_consensus(&vec![message], &self.epoch_store()?)
                 .await?;
         }
+        // todo(zeev): move to func.
```
Comment (zeevmoney) [inline crates/pera-core/src/dwallet_mpc/mpc_manager.rs:329]:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

File: crates/pera-core/src/dwallet_mpc/mpc_manager.rs [hunk 9]
```diff
@@ -435,7 +434,7 @@ impl DWalletMPCManager {
         );
         // TODO (#311): Make sure validator don't mark other validators
         // TODO (#311): as malicious or take any active action while syncing
-        // todo(zeev): remvoed             || !self.pending_instances_queue.is_empty()
+        // todo(zeev): removed             || !self.pending_instances_queue.is_empty()
```
Comment (zeevmoney) [inline crates/pera-core/src/dwallet_mpc/mpc_manager.rs:437]:
```suggestion
```

File: crates/pera-core/src/dwallet_mpc/mpc_outputs_verifier.rs [hunk 10]
```diff
@@ -1,22 +1,26 @@
-///! A module to manage the DWallet MPC outputs.
-/// The module is responsible for storing the outputs received for each instance, and deciding whether an output is valid
-/// by checking if a validators with quorum of stake voted for it.
-/// Any validator that voted for a different output is considered malicious.
+// todo(zeev): fix doc.
```
Comment (zeevmoney) [inline crates/pera-core/src/dwallet_mpc/mpc_outputs_verifier.rs:1]:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

File: crates/pera-core/src/dwallet_mpc/mpc_outputs_verifier.rs [hunk 11]
```diff
@@ -53,34 +58,38 @@ pub struct OutputVerificationResult {
...
+            // Todo (#394): Remove hardcoded network key version.
+            network_key_version,
         }
     }
 
     pub fn network_key_version(&self) -> u8 {
         self.network_key_version
     }
 
+    // todo(zeev): doc
```
Comment (zeevmoney) [inline crates/pera-core/src/dwallet_mpc/mpc_outputs_verifier.rs:92]:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

File: crates/pera-framework/packages/pera-system/sources/dwallet_2pc_mpc_ecdsa_k1.move [hunk 12]
```diff
@@ -791,6 +792,7 @@ module pera_system::dwallet_2pc_mpc_ecdsa_k1 {
         let dwallet_cap_id = object::id(&dwallet_cap);
         transfer::public_transfer(dwallet_cap, tx_context::sender(ctx));
         let session_id = object::id_from_address(tx_context::fresh_object_address(ctx));
+        // todo(zeev): what is this `1`?
```
Comment (zeevmoney) [inline crates/pera-framework/packages/pera-system/sources/dwallet_2pc_mpc_ecdsa_k1.move:795]:
```suggestion
```
use some meaningful name, var if needed.
@ItayLevyOfficial @yaelAbergel

File: crates/pera-framework/packages/pera-system/sources/dwallet_2pc_mpc_ecdsa_k1.move [hunk 13]
```diff
@@ -45,6 +45,7 @@ module pera_system::dwallet_2pc_mpc_ecdsa_k1 {
         message: vector<u8>,
     }
 
+// todo(zeev): fix doc.
```
Comment (zeevmoney) [inline crates/pera-framework/packages/pera-system/sources/dwallet_2pc_mpc_ecdsa_k1.move:48]:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

File: crates/pera-framework/packages/pera-system/sources/dwallet_network_key.move [hunk 14]
```diff
@@ -1,5 +1,7 @@
+// todo(zeev): doc.
```
Comment (zeevmoney) [inline crates/pera-framework/packages/pera-system/sources/dwallet_network_key.move:1]:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

File: crates/pera-framework/packages/pera-system/sources/pera_system.move [hunk 15]
```diff
@@ -597,6 +597,7 @@ module pera_system::pera_system {
         pera_system_state_inner::active_validator_voting_powers(self)
     }
 
+    // todo(zeev): fmt.
```
Comment (zeevmoney) [inline crates/pera-framework/packages/pera-system/sources/pera_system.move:600]:
```suggestion
```

File: crates/pera-graphql-rpc/src/types/transaction_block_kind/dwallet_mpc_output.rs [hunk 16]
```diff
@@ -2,10 +2,12 @@ use crate::types::base64::Base64;
 use async_graphql::Object;
 use pera_types::messages_dwallet_mpc::DWalletMPCOutput as NativeDWalletMPCOutput;
 
+// todo(zeev): doc.
```
Comment (zeevmoney) [inline crates/pera-graphql-rpc/src/types/transaction_block_kind/dwallet_mpc_output.rs:5]:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

File: crates/pera-keys/src/key_derive.rs [hunk 17]
```diff
@@ -186,9 +184,10 @@ pub fn generate_new_key(
     }
 }
 
+// todo(zeev): doc this.
```
Comment (zeevmoney) [inline crates/pera-keys/src/key_derive.rs:187]:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

File: crates/pera-types/src/digests.rs [hunk 18]
```diff
@@ -402,6 +402,7 @@ impl DWalletMPCMessageDigest {
     Deserialize,
     JsonSchema,
 )]
+// todo(zeev): doc.
```
Comment (zeevmoney) [inline crates/pera-types/src/digests.rs:405]:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

File: crates/pera-types/src/transaction.rs [hunk 19]
```diff
@@ -293,10 +293,11 @@ pub enum TransactionKind {
...
     ConsensusCommitPrologueV3(ConsensusCommitPrologueV3),
 
-    /// A transaction with the output of the dwallet MPC flow.
-    /// Used to send the output of the dwallet MPC flow to the other validators so they will be able to
-    /// create a system transaction that writes it to the chain
+    /// A transaction with the output of the dWallet MPC flow.
+    /// Used to send the output of the dwallet MPC flow to the other validators,
+    /// so they will be able to create a system transaction that writes it to the chain
     DWalletMPCOutput(DWalletMPCOutput),
+    // todo(zeev): doc.
```
Comment (zeevmoney) [inline crates/pera-types/src/transaction.rs:300]:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

File: crates/pera-core/src/authority.rs (hunk 1 above)
Comment (ItayLevyOfficial) [reply crates/pera-core/src/authority.rs:1578]: [handled in an open pr](https://github.com/dwallet-labs/dwallet-network/pull/408/commits/74c9758ed8ad319bb07705a33287f65a17b56f2f)

File: crates/pera-core/src/authority/authority_per_epoch_store.rs (hunk 2 above)
Comment (ItayLevyOfficial) [reply crates/pera-core/src/authority/authority_per_epoch_store.rs:1045]: @yaelAbergel

File: crates/pera-core/src/authority/authority_per_epoch_store.rs (hunk 3 above)
Comment (ItayLevyOfficial) [reply crates/pera-core/src/authority/authority_per_epoch_store.rs:1066]: @yaelAbergel

File: crates/pera-core/src/dwallet_mpc/mpc_manager.rs (hunk 9 above)
Comment (ItayLevyOfficial) [reply crates/pera-core/src/dwallet_mpc/mpc_manager.rs:445]: @zeevmoney not sure I understand

File: crates/pera-framework/packages/pera-system/sources/dwallet_network_key.move (hunk 14 above)
Comment (ItayLevyOfficial) [reply crates/pera-framework/packages/pera-system/sources/dwallet_network_key.move:1]: @yaelAbergel

File: crates/pera-keys/src/key_derive.rs (hunk 17 above)
Comment (ItayLevyOfficial) [reply crates/pera-keys/src/key_derive.rs:189]: @yaelAbergel

File: crates/pera-core/src/authority/authority_per_epoch_store.rs (hunk 2 above)
Comment (yaelAbergel) [reply crates/pera-core/src/authority/authority_per_epoch_store.rs:1045]: Changed in later PR

File: crates/pera-core/src/authority/authority_per_epoch_store.rs (hunk 3 above)
Comment (yaelAbergel) [reply crates/pera-core/src/authority/authority_per_epoch_store.rs:1066]: Changed in later PR

File: crates/pera-keys/src/key_derive.rs (hunk 17 above)
Comment (yaelAbergel) [reply crates/pera-keys/src/key_derive.rs:189]: Changed in later PR

**Pull request merged**